controllers/applicationvolumereplication_controller.go

```diff
@@ -417,9 +420,53 @@ func (a *AVRInstance) processPausedSubscription(
 		return IsManifestInAppliedState(pvMW)
 	}
 
+	ready := a.reconciler.isManagedClusterReadyForFailback(subscription, newHomeCluster)
+	a.reconciler.Log.Info(newHomeCluster, "is ready for failback:", ready)
+
+	if !ready {
+		return !unpause
+	}
+
 	return a.cleanupAndRestore(subscription, newHomeCluster)
 }
 
+func (r *ApplicationVolumeReplicationReconciler) isManagedClusterReadyForFailback(
+	subscription *subv1.Subscription, newHomeCluster string) bool {
+	// get VRG and verify status through ManagedClusterView
+	mcvMeta := metav1.ObjectMeta{
+		Name:      "mcv-avr-reconciler",
+		Namespace: newHomeCluster,
+	}
+
+	mcvViewscope := fndv2.ViewScope{
+		Resource:  "VolumeReplicationGroup",
+		Name:      subscription.Name,
+		Namespace: subscription.Namespace,
+	}
+
+	vrg := &rmn.VolumeReplicationGroup{}
+
+	err := r.getManagedClusterResource(mcvMeta, mcvViewscope, vrg)
+	if err != nil {
+		r.Log.Info("getManagedClusterResource failed with error:", err)
+
+		return false
+	}
+
+	r.Log.Info("getManagedClusterResource success")
+
+	return r.isVRGReadyForFailover(vrg)
```

Comment:
The function name in line 433 says ...readyForFailback but in this line you are calling a function that says ...ReadyForFailover.  Is this a typo?

Reply:
Yep, that'd be a typo. `isVRGReadyForFailover` has been refactored to be `isVRGReadyForFailback` on 6338994c1484bc4ee8bbdc8a2ea19d05ca5d2a1f